webapp/src/containers/Rhs/views/ConnectedAccount/ConnectedAccount.container.tsx

```diff
@@ -1,41 +1,42 @@
-import React, {useCallback, useEffect} from 'react';
-import {useDispatch} from 'react-redux';
+import React, {useCallback, useEffect, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
 
 import {Button} from '@brightscout/mattermost-ui-library';
 
-import usePluginApi from 'hooks/usePluginApi';
 import {pluginApiServiceConfigs} from 'constants/apiService.constant';
-import useDialog from 'hooks/useDialog';
+
+import {Dialog} from 'components';
+
+import usePluginApi from 'hooks/usePluginApi';
 import useAlert from 'hooks/useAlert';
 import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+
 import {setConnected} from 'reducers/connectedState';
+
 import {getConnectedState} from 'selectors';
 
+import {ReduxState} from 'types/common/store.d';
+
 import utils from 'utils';
 
 export const ConnectedAccount = () => {
     const dispatch = useDispatch();
+    const [showDialog, setShowDialog] = useState(false);
     const {makeApiRequestWithCompletionStatus, state} = usePluginApi();
     const {username, isAlreadyConnected, msteamsUserId, connected} = getConnectedState(state);
+    const globalState = useSelector((reduxState: ReduxState) => reduxState);
```

Comment:
This makes so any change in the state of the app causes a re-render for this component. Is it intended? Shouldn't we get just the part of the state we need?